### ISSUE TITLE
Auto add better constraint actions

### ISSUE THREAD

**Issue by dereuromark** [cakephp/migrations#385]
This is a (multiple allowed):

* [x] enhancement

* CakePHP Version: latest

### What you did
Deleting a parent of child, that do not have "depending=>true" relationship.
As such, the foreign_key was still the old id, we just removed.
Now there is an invalid foreign key set that doesnt exist as row anymore.

### Expected Behavior
`'delete' => 'SET_NULL'`
should be auto-set for such "NULLable" fields that have no such depending constraint.
This would make sure the foreign_key here is set to NULL again and when we query for all that have no relation to that parent we would also find the right entities.

**Comment by karmicdice** [comment]
This would make sure the foreign_key here is set to NULL again

Tried replicating this behaviour. What I noticed is. If the column is not nullable, it throws an error.

Please elucidate.

**Comment by dereuromark** [comment]
Yes, this is only meant for nullable colums to keep data integrity.
If not nullable, the parent delete must delete also the children (via constraint or domain layer logic / relation setup).

**Comment by dereuromark** [comment]
@markstory Do you think this could be done in a reasonable enough way?
Or is this too complicated?

**Comment by markstory** [comment]
I'm not sure I understand the scenario well. Is the situation that you have model relations defined, but no migration. Then when a migration is generated it should use model relations to infer the schema?

**Comment by LordSimal** [comment]
I am also confused about the scenario here.

As I understand it you already have tables present in your DB, so all you want to do is re-generate the migrations for those tables again with the current recommended config, right?

But it should add the `['delete' => 'SET_NULL']` part of the `addForeignKey()` call the FK column is nullable, right?
Thats at least how I would see it since we primarily use the DB schema to reflect what should get generated.

**Comment by dereuromark** [comment]
Yes, when baking a snapshot, it could fill in those missing things.
But yeah, maybe thats a bit out of scope.

It would also be useful if you add via

    bin/cake bake migration

and the relation is known and "left join" nullable (via null true in db)

**Comment by dereuromark** [comment]
Is this maybe related to https://github.com/cakephp/migrations/issues/318 ?
